.circleci/config.yml

```diff
@@ -0,0 +1,201 @@
+    JAVA_OPTS: -Xmx3200m
+  docker_environment_defaults: &docker_environment_defaults
+    # Customize the JVM to read correct memory values
+    JAVA_OPTS: '-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -XX:MaxRAMFraction=1'
+
+  default_java_job: &default_java_job
+    docker:
+      - image: circleci/openjdk:8-jdk
+        environment:
+          <<: *docker_environment_defaults
+    environment:
+      <<: *host_environment_defaults
+    working_directory: ~/workdir
+
+  restore_source_code: &restore_source_code
+    restore_cache:
+      keys:
+        - source-code-0.0.1-{{ .Branch }}-{{ .Revision }}
+        - source-code-0.0.1-{{ .Branch }}
+        - source-code-0.0.1
+  restore_dependencies: &restore_dependencies
+    restore_cache:
+      keys:
+        - sbt-ivy2-dependencies-0.0.1-{{ checksum "/tmp/dependencies.cache.tmp" }}
+        - sbt-ivy2-dependencies-0.0.1
+  restore_build: &restore_build
+    restore_cache:
+      keys:
+        - sbt-build-0.0.1-{{ .Branch }}-{{ .Revision }}
+        - sbt-build-0.0.1
+
+  setup_dependencies_key: &setup_dependencies_key
+    run:
+      name: Generate cache key
+      command: |
+        shasum build.sbt \
+          project/plugins.sbt \
+          project/build.properties \
+          project/Common.scala \
+          project/Dependencies.scala > /tmp/dependencies.cache.tmp
+
+jobs:
+  checkout_code:
+    <<: *default_java_job
+    steps:
+      - *restore_source_code
+      - checkout
+      - save_cache:
+          key: source-code-0.0.1-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - ~/workdir
+
+  sbt_dependencies:
+    <<: *default_java_job
+    steps:
+      - *restore_source_code
+      - *setup_dependencies_key
+      - *restore_dependencies
+      - run:
+          name: Resolve dependencies
+          command: |
+            sbt update
+      - save_cache:
+          paths:
+            - ~/.ivy2
+            - ~/.sbt
+          key: sbt-ivy2-dependencies-0.0.1-{{ checksum "/tmp/dependencies.cache.tmp" }}
+
+  compile:
+    <<: *default_java_job
+    steps:
+      - *restore_source_code
+      - *setup_dependencies_key
+      - *restore_dependencies
+      - *restore_build
+      - run:
+          name: Compile
+          command: ./scripts/compile.sh
+      - save_cache:
+          paths:
+            - ~/workdir/target
+            - ~/workdir/project/target
+          key: sbt-build-0.0.1-{{ .Branch }}-{{ .Revision }}
+
+  test:
+    <<: *default_java_job
+    steps:
+      - *restore_source_code
+      - *setup_dependencies_key
+      - *restore_dependencies
+      - *restore_build
+      - run:
+          name: Install the tool
+          command: |
+            apt-get update
+            sudo apt-add-repository -y ppa:brightbox/ruby-ng
+            sudo apt-get -y update
+            sudo apt-get -y install ruby2.2 ruby2.2-dev
+            gem install rake hoe sexp_processor ruby_parser ruby2ruby erubis
```

Comment:
This should be installed from a Gemfile. Check https://github.com/codacy/codacy-rubocop/commit/ef1e81597aada33de01992255178d5fbe3f0640a for an example.